packages/provider/src/transports/wallet-request-handler.ts

```diff
@@ -796,6 +827,18 @@ export class WalletRequestHandler implements ExternalProvider, JsonRpcHandler, P
     }
     return false
   }
+
+  private saveConnectedDapp(connectedDapp: ConnectedDapp) {
+    const connectedDappsKey = LOCAL_STORAGE_KEYS.CONNECTED_DAPPS
+    const data = window.localStorage.getItem(connectedDappsKey)
+    const connectedDapps: ConnectedDapp[] = data ? JSON.parse(data) : []
+    if (!connectedDapps.map(dapp => dapp.origin).includes(connectedDapp.origin)) {
```

Comment:
I think I'd reverse the order of the if-else blocks here and do `connectedDapps.some(dapp => dapp.origin === connectedDapp.origin)`, but feel free to ignore since it's really minor.